nltk/test/unit/translate/test_bleu.py

```diff
@@ -122,7 +122,9 @@ def test_full_matches(self):
     def test_partial_matches_hypothesis_longer_than_reference(self):
         references = ['John loves Mary'.split()]
         hypothesis = 'John loves Mary who loves Mike'.split()
-        self.assertAlmostEqual(sentence_bleu(references, hypothesis), 0.4729, places=4)
+        # Since no 4-grams matches were found the result should be zero
+        # exp(w_1 * 1 * w_2 * 1 * w_3 * 1 * w_4 * -inf) = 0
+        self.assertAlmostEqual(sentence_bleu(references, hypothesis), 0.0, places=4)
```

Comment:
This makes little sense to give 0 credits to this hypothesis given the references. 

The sentence is half correct, saying that it's 0.0 BLEU is really not sound at all.